var/spack/repos/builtin/packages/netlib-scalapack/package.py

```diff
@@ -14,12 +14,18 @@ class ScalapackBase(CMakePackage):
 
     variant("shared", default=True, description="Build the shared library version")
     variant("pic", default=False, description="Build position independent code")
+    variant(
+        "ilp64",
+        default=False,
+        description="Build with 8-byte (long) integers rather than the regular 4-byte ones",
+    )
 
     provides("scalapack")
 
     depends_on("mpi")
     depends_on("lapack")
     depends_on("blas")
+    depends_on("openblas+ilp64", when="+ilp64")  # virtuals and variants do not play well together
```

Comment:
`quantum-espresso` has an example of how to do this in a way that does not tie this package specifically to openblas:

```
requires("^fftw+openmp", when="^[virtuals=fftw-api] fftw")
```

in this case it would look like:

```
with when("+ilp64"):
    requires("^openblas+ilp64", when="^[virtuals=blas] openblas")
```